pkg/webservice/handlers.go

```diff
@@ -224,6 +226,42 @@ func getApplicationJSON(app *objects.Application) *dao.ApplicationDAOInfo {
 		}
 		allocationInfo = append(allocationInfo, allocInfo)
 	}
+	for _, req := range requests {
```

Comment:
Move all this processing into its own function. Do the sorting in there as per comment earlier around unlocked sorting.

Reply:
Done.